### PR TITLE
[REVIEW] Update to arrow-cpp and pyarrow 0.15.0

### PR DESCRIPTION
**Pull request by trxcllnt**
This PR updates arrow-cpp and pyarrow to v0.15.0.

On the pyarrow side, the most significant change is the removal of the `pa.Column` class, and `pa.Array.to_pandas()` now returning a `pd.Series` instead of an `np.NDArray`.

arrow-cpp v0.15.0 fixes the following nvcc compiler warnings:
```
arrow/install/include/arrow/io/interfaces.h(195): warning: attribute does not apply to any entity
```
but it brings in these new ones (tracked in [ARROW-6886](https://issues.apache.org/jira/browse/ARROW-6886)):
```
arrow/install/include/arrow/io/file.h(189): warning: overloaded virtual function "arrow::io::Writable::Write" is only partially overridden in class "arrow::io::MemoryMappedFile"

arrow/install/include/arrow/io/memory.h(98): warning: overloaded virtual function "arrow::io::Writable::Write" is only partially overridden in class "arrow::io::MockOutputStream"

arrow/install/include/arrow/io/memory.h(116): warning: overloaded virtual function "arrow::io::Writable::Write" is only partially overridden in class "arrow::io::FixedSizeBufferWriter"

arrow/install/include/arrow/io/file.h(189): warning: overloaded virtual function "arrow::io::Writable::Write" is only partially overridden in class "arrow::io::MemoryMappedFile"

arrow/install/include/arrow/io/memory.h(98): warning: overloaded virtual function "arrow::io::Writable::Write" is only partially overridden in class "arrow::io::MockOutputStream"

arrow/install/include/arrow/io/memory.h(116): warning: overloaded virtual function "arrow::io::Writable::Write" is only partially overridden in class "arrow::io::FixedSizeBufferWriter"

arrow/install/include/arrow/io/file.h(189): warning: overloaded virtual function "arrow::io::Writable::Write" is only partially overridden in class "arrow::io::MemoryMappedFile"

arrow/install/include/arrow/io/memory.h(98): warning: overloaded virtual function "arrow::io::Writable::Write" is only partially overridden in class "arrow::io::MockOutputStream"

arrow/install/include/arrow/io/memory.h(116): warning: overloaded virtual function "arrow::io::Writable::Write" is only partially overridden in class "arrow::io::FixedSizeBufferWriter"
```

Related PR to remove pyarrow dependencies from cugraph: https://github.com/rapidsai/cugraph/pull/562

### CODE REVIEW

**Comment by trxcllnt** [comment]
rerun tests

File: python/cudf/cudf/core/column/string.py [hunk 1]
```diff
@@ -586,7 +586,7 @@ def to_arrow(self):
             )
 
     def to_pandas(self, index=None):
-        pd_series = self.to_arrow().to_pandas()
+        pd_series = self.to_arrow().to_pandas().array
```
[the remark this answers is not in the document]
Comment (trxcllnt) [reply python/cudf/cudf/core/column/string.py]: I didn't see a Pandas API to set the index on an existing Series, and creating a Series with a new index from the old Series essentially reindexes the old Series (which is wrong in certain cases). pyarrow used to return a Numpy Array from `to_pandas()` where now it returns a pd.Series, so this change makes the logic consistent with v0.15 Arrow.

**Comment by trxcllnt** [comment]
@kkraus14 given that 0.15 makes changes in the python Arrow APIs, I'd lean on merging this sooner. Arrow 0.15.1 will go out in the next week or two, and updating to that to remove the compiler warnings should be a much easier task if this is already merged.

**Comment by trxcllnt** [comment]
I should mention, cuGraph will need to merge https://github.com/rapidsai/cugraph/pull/562 before/or shortly after we merge this.

File: python/cudf/cudf/core/column/string.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kkraus14) [reply python/cudf/cudf/core/column/string.py]: You can just do `pd_series.index = new_index` and `pd_series.name = new_name`

[the remark this answers is not in the document]
Comment (trxcllnt) [reply python/cudf/cudf/core/column/string.py]: Thanks @kkraus14, I didn't know those were publicly mutable. Just pushed a commit with the changes.

**Comment by kkraus14** [comment]
Depends on https://github.com/rapidsai/cugraph/pull/562

**Pull request merged**